gr-fec/lib/fec_mtrx_impl.cc

```diff
@@ -148,39 +142,43 @@ void write_matrix_to_file(const std::string filename, matrix_sptr M)
 
 matrix_sptr generate_G_transpose(matrix_sptr H_obj)
 {
-    unsigned int k = H_obj->size1;
-    unsigned int n = H_obj->size2;
-    unsigned int row_index, col_index;
+    const unsigned int k = H_obj->size1;
+    const unsigned int n = H_obj->size2;
 
-    gsl_matrix* G_transp = gsl_matrix_alloc(n, k);
+    gsl_matrix_uchar* G_transp = gsl_matrix_uchar_alloc(n, k);
 
     // Grab P' matrix (P' denotes P transposed)
-    gsl_matrix* P_transpose = gsl_matrix_alloc(n - k, k);
-    for (row_index = 0; row_index < n - k; row_index++) {
-        for (col_index = 0; col_index < k; col_index++) {
-            int value = gsl_matrix_get((gsl_matrix*)(H_obj.get()), row_index, col_index);
-            gsl_matrix_set(P_transpose, row_index, col_index, value);
+    gsl_matrix_uchar* P_transpose = gsl_matrix_uchar_alloc(n - k, k);
+    for (unsigned int row_index = 0; row_index < n - k; row_index++) {
+        for (unsigned int col_index = 0; col_index < k; col_index++) {
+            gsl_matrix_uchar_set(P_transpose,
+                                 row_index,
+                                 col_index,
+                                 gsl_matrix_uchar_get(H_obj.get(), row_index, col_index));
```

Comment:
So despite what the comment says, there isn't a transpose operation being done here (the comment might still make sense depending on how the notation for the blocks of the generator and parity check matrix is defined).

This code is basically copying out a submatrix: `P_transpose[row_index, col_index] = H_obj[row_index, col_index]`. Might there be a way to write this more compactly with GSL functions? From a quick look at the [docs](https://www.gnu.org/software/gsl/doc/html/vectors.html), I don't see it, but it also seems to me that copying into `P_transpose` is not necessary. A "matrix view" might be enough, or even using `H_obj` instead.